b2sdk/utils.py

```diff
@@ -366,6 +366,24 @@ class B2TraceMetaAbstract(DefaultTraceAbstractMeta):
     pass
 
 
+class ConcurrentUsedAuthTokenGuard(object):
+    def __init__(self, lock_dict, lock_key):
+        self.lock_dict = lock_dict
+        self.lock_key = lock_key
+
+    def __enter__(self):
+        if not self.lock_dict[self.lock_key].acquire(False):
+            from b2sdk.exception import UploadTokenUsedConcurrently
```

Comment:
do we really need to import it right here?

Reply:
@ppolewicz yes; otherwise there would be an import loop:

```
Traceback (most recent call last):
  File "/Users/vmax/.virtualenvs/b2sdk/lib/python3.7/site-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/Users/vmax/.virtualenvs/b2sdk/lib/python3.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/Users/vmax/.virtualenvs/b2sdk/lib/python3.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/Users/vmax/.virtualenvs/b2sdk/lib/python3.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/Users/vmax/.virtualenvs/b2sdk/lib/python3.7/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/Users/vmax/.virtualenvs/b2sdk/lib/python3.7/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/vmax/work/Reef/b2-sdk-python/test/v1/test_utils.py", line 15, in <module>
    from .deps import b2_url_encode, b2_url_decode, choose_part_ranges, format_and_scale_number, format_and_scale_fraction
  File "/Users/vmax/work/Reef/b2-sdk-python/test/v1/deps.py", line 11, in <module>
    from b2sdk.v1 import *
  File "/Users/vmax/work/Reef/b2-sdk-python/b2sdk/v1/__init__.py", line 16, in <module>
    from b2sdk.api import B2Api
  File "/Users/vmax/work/Reef/b2-sdk-python/b2sdk/api.py", line 13, in <module>
    from .account_info.sqlite_account_info import SqliteAccountInfo
  File "/Users/vmax/work/Reef/b2-sdk-python/b2sdk/account_info/__init__.py", line 11, in <module>
    from .in_memory import InMemoryAccountInfo
  File "/Users/vmax/work/Reef/b2-sdk-python/b2sdk/account_info/in_memory.py", line 11, in <module>
    from .exception import MissingAccountData
  File "/Users/vmax/work/Reef/b2-sdk-python/b2sdk/account_info/exception.py", line 15, in <module>
    from ..exception import B2Error
  File "/Users/vmax/work/Reef/b2-sdk-python/b2sdk/exception.py", line 17, in <module>
    from .utils import camelcase_to_underscore
  File "/Users/vmax/work/Reef/b2-sdk-python/b2sdk/utils.py", line 23, in <module>
    from b2sdk.exception import UploadTokenUsedConcurrently
ImportError: cannot import name 'UploadTokenUsedConcurrently' from 'b2sdk.exception' (/Users/vmax/work/Reef/b2-sdk-python/b2sdk/exception.py)


```